include/spblas/vendor/onemkl_sycl/detail/get_matrix_handle.hpp

```diff
@@ -18,14 +18,16 @@ template <matrix M>
 oneapi::mkl::sparse::matrix_handle_t
 get_matrix_handle(sycl::queue& q, M&& m,
                   oneapi::mkl::sparse::matrix_handle_t handle = nullptr) {
-  if constexpr (__detail::is_matrix_opt_view_v<decltype(m)>) {
+  if constexpr (__detail::is_matrix_opt_v<decltype(m)>) {
     log_trace("using A as matrix_opt");
 
     if (m.matrix_handle_ == nullptr) {
       m.matrix_handle_ = create_matrix_handle(q, m.base());
     }
 
     return m.matrix_handle_;
+  } else if constexpr (__detail::has_base<M>) {
+    return get_matrix_handle(q, m.base(), handle);
```

Comment:
oh, so line 29 and 30 will iterate through the stack and keep looking until we either find a matrix_handle, or we run out of bases and check either handle or create_matrix_handle on ultimate base of m ...